File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        int leftIndex = 0;
        int rightIndex = nums.length - 1;

        while (leftIndex < rightIndex) {
            int middleIndex = (leftIndex + (rightIndex - leftIndex) / 2);

            if (nums[middleIndex] < nums[rightIndex]) {
                // minimum is on the left half
                rightIndex = middleIndex;
            } else {
                // minimum is in the right half 
                leftIndex = middleIndex + 1;
            }
        }
        return nums[leftIndex];
    }
}

File: Data Structures & Algorithms/monotonic-array/submission-0.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        diff_set = set()
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                diff_set.add(1)
            elif nums[i] < nums[i-1]:
                diff_set.add(-1)
            else:
                diff_set.add(0)
        
        if 1 in diff_set and -1 in diff_set:
            return False
        return True